cooked-validators/src/Cooked/MockChain/UtxoState.hs

```diff
@@ -12,35 +12,44 @@ import qualified Plutus.V1.Ledger.Value as Pl1
 
 -- | A 'UtxoState' provides us with the mental picture of the state of the UTxO graph:
 -- Each address has a set of UTxOs that consist in a value and some potential datum.
-newtype UtxoState = UtxoState {utxoState :: Map Pl.Address UtxoValueSet}
+newtype UtxoState = UtxoState {utxoState :: Map Pl.Address UtxoPayloadSet}
   deriving (Eq)
 
 instance Semigroup UtxoState where
   (UtxoState a) <> (UtxoState b) = UtxoState $ Map.unionWith (<>) a b
 
--- | Represents a /set/ of values, yet, we use a list instead of a set because 'Pl.Value'
--- doesn't implement 'Ord' and because it is possible that we want to distinguish between utxo states
--- that have additional utxos, even if these could have been merged together.
-newtype UtxoValueSet = UtxoValueSet {utxoValueSet :: [(Pl1.Value, TxSkelOutDatum)]}
+-- | Represents a /set/ of payloads. Payloads are the name we give to the
+-- information we care about on a UTxO: value, datum, and reference script. We
+-- use a list instead of a set because 'Pl.Value' doesn't implement 'Ord' and
+-- because it is possible that we want to distinguish between utxo states that
+-- have additional utxos, even if these could have been merged together.
+newtype UtxoPayloadSet = UtxoPayloadSet {utxoPayloadSet :: [UtxoPayload]}
   deriving (Show)
 
-instance Eq UtxoValueSet where
-  (UtxoValueSet xs) == (UtxoValueSet ys) = xs' == ys'
+data UtxoPayload = UtxoPayload
+  { utxoPayloadValue :: Pl1.Value,
+    utxoPayloadSkelOutDatum :: TxSkelOutDatum,
+    utxoPayloadReferenceScript :: Maybe Pl.ScriptHash
```

Comment:
One thing I've seen folks do to avoid long accessor names is to abbreviate like `upValue`, `upSkelOutDatum`, etc. Entirely up to you whether you want to do that, of course!

Reply:
I do tend to be a bit too verbose when naming things. I am not against shortening the prefix in those accessor names at all but I can't come up with anything else and I think I would be confused if I saw names such as `upValue` in the code in a few months and would not think about `UtxoPayload`. The `UtxoPayload` type is not used a lot at all nor very important (`UtxoState` is only used for pretty printing and test predicates). I would rather keep the longer name for now although I agree.

As for importing modules qualified, maybe it is a more general topic and deserves a team-wide discussion in an issue. On a related note, @carlhammann and I had briefly considered using export lists as well in our modules. There were upsides and downsides. In the end we decided to stick with the status quo where everything is exported, and cooked modules imported unqualified.